### PR TITLE
fix: assert group(partition) key type during physical plan compilation

### LINKED ISSUE #613: window partition by Miss index execution result error when performance is not sensitive
partition by float or double result error,but  partition by int is right
## Expected Behavior
 ---- ---- ---------- -----------
  id   c1   c5         w1_c4_sum
 ---- ---- ---------- -----------
  1    aa   1.100000   30
  2    bb   1.100000   61
  3    cc   1.100000   93
  4    dd   1.100000   96
  5    ee   1.200000   34
 ---- ---- ---------- -----------
## Current Behavior
 ---- ---- ---------- -----------
  id   c1   c5         w1_c4_sum
 ---- ---- ---------- -----------
  1    aa   1.100000   30
  2    bb   1.100000   61
  3    cc   1.100000   93
  4    dd   1.100000   96
  5    ee   1.200000   99
 ---- ---- ---------- -----------

## Steps to Reproduce
1. create
create table auto_LfLMDwcY(
id int,
c1 string,
c3 int,
c4 bigint,
c5 float,
c6 double,
c7 timestamp,
c8 date,
index(key=(c1),ts=c7))options(partitionnum=1,replicanum=1);
2. insert
insert into auto_LfLMDwcY values(1,'aa',20,30,1.1,2.1,1590738990000,'2020-05-01');
insert into auto_LfLMDwcY values(2,'bb',20,31,1.1,2.2,1590738991000,'2020-05-02');
insert into auto_LfLMDwcY values(3,'cc',20,32,1.1,2.3,1590738992000,'2020-05-03');
insert into auto_LfLMDwcY values(4,'dd',20,33,1.1,2.4,1590738993000,'2020-05-04');
insert into auto_LfLMDwcY values(5,'ee',21,34,1.2,2.5,1590738994000,'2020-05-05');
3. set performance_sensitive
set performance_sensitive=false;
4. select
SELECT id, c1, c5, sum(c4) OVER w1 as w1_c4_sum FROM auto_LfLMDwcY WINDOW w1 AS (PARTITION BY auto_LfLMDwcY.c5 ORDER BY auto_LfLMDwcY.c7 ROWS_RANGE BETWEEN 2s PRECEDING AND CURRENT ROW);

## case
```yaml
  - id: 2
    desc: rows_range-float为partition by-未命中索引
    inputs:
      - columns: [ "id int","c1 string","c3 int","c4 bigint","c5 float","c6 double","c7 timestamp","c8 date" ]
        indexs: [ "index1:c1:c7" ]
        rows:
          - [ 1,"aa",20,30,1.1,2.1,1590738990000,"2020-05-01" ]
          - [ 2,"bb",20,31,1.1,2.2,1590738991000,"2020-05-02" ]
          - [ 3,"cc",20,32,1.1,2.3,1590738992000,"2020-05-03" ]
          - [ 4,"dd",20,33,1.1,2.4,1590738993000,"2020-05-04" ]
          - [ 5,"ee",21,34,1.2,2.5,1590738994000,"2020-05-05" ]
    sql: |
      SELECT id, c1, c5, sum(c4) OVER w1 as w1_c4_sum FROM {0} WINDOW w1 AS (PARTITION BY {0}.c5 ORDER BY {0}.c7 ROWS_RANGE BETWEEN 2s PRECEDING AND CURRENT ROW);
    expect:
      order: id
      columns: [ "id int","c1 string","c5 float","w1_c4_sum bigint" ]
      rows:
        - [ 1,"aa",1.1,30 ]
        - [ 2,"bb",1.1,61 ]
        - [ 3,"cc",1.1,93 ]
        - [ 4,"dd",1.1,96 ]
        - [ 5,"ee",1.2,34 ]
```

### PR DESCRIPTION
**Pull request by aceforeverd**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

fix: #613 

in `performance sensitive mode` = false, group key type is not checked. And, allow unsupport key like float as partition key.

Basicly, the runner only support partition key is
- bool
- intxxx
- string
- date
- timestamp

any other type will result compilation error

### CODE REVIEW

File: hybridse/src/passes/physical/group_and_sort_optimized.cc [hunk 1]
```diff
@@ -454,6 +459,27 @@ bool GroupAndSortOptimized::TransformKeysAndOrderExpr(
                                                      &source_column_name)) {
                     return false;
                 }
+
+                size_t schema_idx = 0;
+                size_t col_idx = 0;
+                if (root_schemas_ctx->ResolveColumnRefIndex(column, &schema_idx, &col_idx).isOK()) {
```
Comment (aceforeverd) [inline hybridse/src/passes/physical/group_and_sort_optimized.cc]: window & last join & group

Comment (aceforeverd) [reply hybridse/src/passes/physical/group_and_sort_optimized.cc]: Status BatchModeTransformer::CheckWindow
do not add in group_and_sort_optimized

File: hybridse/include/sdk/base.h [hunk 2]
```diff
@@ -72,6 +73,34 @@ inline const std::string DataTypeName(const DataType& type) {
...
+        case type::kBool:
+            return "bool";
+        case type::kBlob:
+            return "blob";
+        case type::kNull:
+            return "null";
+    }
+    return "unknown";
+}
+
```
Comment (jingchen2222) [inline hybridse/include/sdk/base.h]: We won't expose fe_type.ph.h to SDK users. That's why we need `sdk::DataType`. 
Move `inline const std::string TypeName(type::Type type) ` to somewhere else like include/node/type_node.h or include/node/sql_node.h

BTW, we are going to refactor prototypes in hybridse and openmldb, which means this function might be rewritten in the future.  So we don't have to worry too much about if it's appropriate to place `TypeName (type::Type type)` in include/node/xxx.h.

File: hybridse/src/vm/transform.cc [hunk 3]
```diff
@@ -1751,6 +1753,45 @@ Status BatchModeTransformer::CheckTimeOrIntegerOrderColumn(const node::OrderByNo
...
+                        case type::kBool:
+                        case type::kInt16:
+                        case type::kInt32:
+                        case type::kInt64:
+                        case type::kVarchar:
+                        case type::kDate:
+                        case type::kTimestamp:
+                            break;
+                        default: {
+                            CHECK_TRUE(false, common::kPhysicalPlanError, "unsupported group key, type is ",
```
Comment (jingchen2222) [inline hybridse/src/vm/transform.cc]: Use FAIL_STATUS(common::kPhysicalPlanError, msg) instead.

File: hybridse/src/passes/physical/group_and_sort_optimized.cc [hunk 4]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include "passes/physical/group_and_sort_optimized.h"
+#include <absl/strings/str_cat.h>
```
Comment (jingchen2222) [inline hybridse/src/passes/physical/group_and_sort_optimized.cc]: unused, remove?

File: hybridse/include/sdk/base.h (hunk 2 above)
Comment (aceforeverd) [reply hybridse/include/sdk/base.h]: cool. BTW there is `sdk::Type_Name` provided by protobuf, but I'd like to rm the `k` prefix

**Review by jingchen2222 — APPROVED** [review]
LGTM

**Comment by jingchen2222** [comment]
Pls fix https://github.com/4paradigm/OpenMLDB/runs/4229503341?check_suite_focus=true

**Review by jingchen2222 — APPROVED** [review]
LGTM

**Review by vagetablechicken — APPROVED** [review]
LGTM

**Comment by lumianph** [comment]
[bugfix tracking](https://trello.com/c/wEQ9HW0k/25-bugfix-tracking)

**Pull request merged**